docs/scratch-deploy.md

```diff
@@ -0,0 +1,189 @@
+
+```shell
+NETWORK=<PUT-YOUR-VALUE> RPC_URL=<PUT-YOUR-VALUE> bash ./scripts/scratch/verify-contracts-code.sh
+```
+
+### Issues with verification of part of the contracts deployed from factories
+
+There are some contracts deployed from other contracts for which automatic hardhat etherscan verification fails:
+
+- `AppProxyUpgradeable` of multiple contracts (`app:lido`, `app:node-operators-registry`, `app:oracle`, `app:voting`, ...)
+- `KernelProxy` -- proxy for `Kernel`
+- `AppProxyPinned` -- proxy for `EVMScriptRegistry`
+- `MiniMeToken` -- LDO token
+- `CallsScript` -- Aragon internal contract
+- `EVMScriptRegistry` -- Aragon internal contract
+
+The workaround used during Holešky deployment is to deploy auxiliary instances of these contracts standalone and verify them via hardhat Etherscan plugin. After this Etherscan will mark the target contracts as verified by "Similar Match Source Code".
+
+NB, that some contracts require additional auxiliary contract to be deployed. Namely, constructor of `AppProxyPinned` depends on proxy implementation ("base" in Aragon terms) contract with `initialize()` function and `Kernel` contract which must return the implementation by call `kernel().getApp(KERNEL_APP_BASES_NAMESPACE, _appId)`. See `@aragon/os/contracts/apps/AppProxyBase.sol` for the details.
+
+## Post deploy initialization
+
+### Initialization up to fully operational state
+
+In order to make protocol fully operational the additional steps are required.
+
+- add `NodeOperatorsRegistry` as staking module: `StakingRouter.addStakingModule`
+- add oracle committee members to `HashConsensus` contracts for `AccountingOracle` and `ValidatorsExitBusOracle`: `HashConsensus.addMember`
+- initialize initial epoch for `HashConsensus` contracts for `AccountingOracle` and `ValidatorsExitBusOracle`: `HashConsensus.updateInitialEpoch`
+- add guardians to `DepositSecurityModule`: `DepositSecurityModule.addGuardians`
+- resume protocol: `Lido.resume`
+- resume WithdrawalQueue: `WithdrawalQueueERC721.resume`
+- add at least one Node Operator: `NodeOperatorsRegistry.addNodeOperator`
+- add validator keys to the Node Operators: `NodeOperatorsRegistry.addSigningKeys`
+- set staking limits for the Node Operators: `NodeOperatorsRegistry.setNodeOperatorStakingLimit`
+
+NB, that part of the actions require preliminary granting of the required roles, e.g. `STAKING_MODULE_MANAGE_ROLE` for `StakingRouter.addStakingModule`:
+
+```js
+  await stakingRouter.grantRole(STAKING_MODULE_MANAGE_ROLE, agent.address, { from: agent.address })
+  await stakingRouter.addStakingModule(
+    state.nodeOperatorsRegistry.deployParameters.stakingModuleTypeId,
+    nodeOperatorsRegistry.address,
+    NOR_STAKING_MODULE_TARGET_SHARE_BP,
+    NOR_STAKING_MODULE_MODULE_FEE_BP,
+    NOR_STAKING_MODULE_TREASURY_FEE_BP,
+    { from: agent.address }
+  )
+  await stakingRouter.renounceRole(STAKING_MODULE_MANAGE_ROLE, agent.address, { from: agent.address })
+```
+
+## Protocol parameters
```

Comment:
Have you checked the final ACL setup against the https://docs.lido.fi/guides/verify-lido-v2-upgrade-manual#roles-setup ? (esp. Burner-related roles for NOR and Lido)